docker-entrypoint/docker-entrypoint.go

```diff
@@ -0,0 +1,208 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+	//For testing purposes
+	//"k8s.io/kubernetes/pkg/client/restclient"
+
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+const (
+	WAITING  = "waiting"
+	READY    = "ready"
+	RUNNING  = "Running"
+	INTERVAL = 2
+)
+
+var (
+	//"Info logger""
+	Info *log.Logger
+	//"Error logger"
+	Error *log.Logger
+)
+
+type dependency interface {
+	Exists(namespace string, name string) error
+	DepResolved(namespace string, name string) (bool, error)
+	GetType() string
+}
+
+func WaitFor(dep dependency, namespace string, names []string) error {
+	depState := WAITING
+	if names == nil {
+		depState = READY
+		Info.Println("Container has no ", dep.GetType(), " dependencies")
+	}
+
+	for depState == WAITING {
+
+		for _, name := range names {
```

Comment:
Well, should we keep testing resolved 100 deps because 1 isn't ready yet? I am not sure it is a requirement of this tool. Correct me if I'm wrong. Instead, I would make one big list of deps in main, pass it to `WaitFor([]Dependency, namespace`, and then initialize a sync.WaitGroup, start one go routine per dep (add it to wait group) that would: test DepResolved, if true: wg.Done() and return; if false: backoff (maybe even exponential backoff with a quite short upper wait time). That would be idiomatic.